satellite/satellitedb/bandwidthagreement.go

```diff
@@ -85,23 +79,41 @@ func (b *bandwidthagreement) GetTotals(ctx context.Context, from, to time.Time)
 
 	totals := make(map[storj.NodeID][]int64)
 	for i := 0; rows.Next(); i++ {
-		var nodeID []byte
+		var nodeID storj.NodeID
 		data := make([]int64, len(pb.BandwidthAction_value))
 		err := rows.Scan(&nodeID, &data[pb.BandwidthAction_PUT], &data[pb.BandwidthAction_GET],
 			&data[pb.BandwidthAction_GET_AUDIT], &data[pb.BandwidthAction_GET_REPAIR], &data[pb.BandwidthAction_PUT_REPAIR])
 		if err != nil {
 			return totals, err
 		}
-		id, err := storj.NodeIDFromBytes(nodeID)
+		totals[nodeID] = data
+	}
+	return totals, nil
+}
+
+//GetExpired gets orders that are expired and were created before some time
+func (b *bandwidthagreement) GetExpired(before time.Time, expiredAt time.Time) (orders []bwagreement.SavedOrder, err error) {
```

Comment:
Do we need `before` in `GetExpired` and `DeleteExpired`? Is there a case where we don't want to delete something that is expired because it was not created before some time?

Reply:
Yes.  We need to make sure we ran the accounting tally before we delete expired agreements.